src/wasm-traversal.h

```diff
@@ -157,185 +149,209 @@ struct WasmWalkerBase : public WasmVisitor<SubType, ReturnType> {
     self->visitMemory(&module->memory);
     self->visitModule(module);
   }
-};
 
-template<typename ParentType>
-struct ChildWalker : public WasmWalkerBase<ChildWalker<ParentType>> {
-  ParentType& parent;
+  // Walk implementation. We don't use recursion as ASTs may be highly
+  // nested.
 
-  ChildWalker(ParentType& parent) : parent(parent) {}
+  // Tasks receive the this pointer and a pointer to the pointer to operate on
+  typedef void (*TaskFunc)(SubType*, Expression**);
```

Comment:
Is `visitExpression` allowed to replace the current node? We should probably document it either way.